### PR TITLE
EZP-30333: The host's port is added to image variations URIs when configured through image_host

### PR DESCRIPTION
**Pull request by kmadejski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30333](https://jira.ez.no/browse/EZP-30333)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `6.7`/`6.13`/`7.4`/`7.5`/`master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Please take a look at JIRA ticket for more information. 

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Cache\Resolver;
+
+use Liip\ImagineBundle\Imagine\Cache\Resolver\ProxyResolver as ImagineProxyResolver;
+
+class ProxyResolver extends ImagineProxyResolver
+{
+    /**
+     * Replaces host with given proxy host.
+     *
```
Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php]: maybe add some doc here on why this method is overloaded. maybe even short diff compared to liip for future maintenance know-how.

File: eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     * @param $url
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php]:
```suggestion
     * @param string $url
```

File: eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php (hunk 1 above)
Comment (kmadejski) [reply eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/Resolver/ProxyResolver.php]: Added short info in https://github.com/ezsystems/ezpublish-kernel/pull/2604/commits/5af6ddf46b4befbb8e262d9b90f07b4966fdcd73.

**Comment by andrerom** [comment]
@lserwatka Candidate for 2.5.1

**Comment by lserwatka** [comment]
@andrerom we know this, @mnocon is going to look into this.

**Pull request merged**